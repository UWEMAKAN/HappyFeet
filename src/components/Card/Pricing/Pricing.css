.Pricing {
  width: 100%;
  align-self: stretch;
  display: flex;
}

.Pricing span:not(:last-child) {
  margin-right: 1rem;
}

.Pricing span:last-child {
  color: #209CEE;
}

.Pricing span:nth-child(2) {
  text-decoration: line-through;
  color: #aaa;
}

