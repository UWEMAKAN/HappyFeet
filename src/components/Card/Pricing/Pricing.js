import React from 'react';
import propTypes from 'prop-types';
import './Pricing.css';

const Pricing = ({ price, discount }) => {
  const originalPrice = Math.ceil(price / (1 - (discount / 100)));
  const formatter = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'INR',
  });

  return (
    <div className="Pricing">
      <span>{`${formatter.format(price)}`.split('.')[0]}</span>
      <span>{`${formatter.format(originalPrice)}`.split('.')[0]}</span>
      <span>{`${discount}% off`}</span>
    </div>
  );
};

Pricing.propTypes = {
  price: propTypes.number.isRequired,
  discount: propTypes.number.isRequired
};

export default Pricing;
