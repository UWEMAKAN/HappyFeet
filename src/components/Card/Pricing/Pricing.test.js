import React from 'react';
import { shallow } from 'enzyme';
import Pricing from './Pricing';

const renderPricing = (args) => {
  const defaultProps = {
    price: 0,
    discount: 0
  };
  const props = { ...defaultProps, ...args };
  // eslint-disable-next-line react/jsx-props-no-spreading
  return shallow(<Pricing {...props} />);
};

describe('Pricing', () => {
  test('should render Pricing Component', () => {
    const price = 3500;
    const discount = 30;
    const wrapper = renderPricing({ price, discount });
    expect.assertions(1);
    expect(wrapper).toMatchSnapshot();
  });
});
