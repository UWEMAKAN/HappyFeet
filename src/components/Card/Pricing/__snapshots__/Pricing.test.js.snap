// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pricing should render Pricing Component 1`] = `ShallowWrapper {}`;
